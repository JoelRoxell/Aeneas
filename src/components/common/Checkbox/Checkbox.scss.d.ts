export const checkbox: string;
export const active: string;
