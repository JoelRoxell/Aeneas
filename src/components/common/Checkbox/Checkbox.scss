.checkbox {
  color: white;
  height: rem(20);
  width: rem(20);
  display: flex;
  align-items: center;
  justify-content: center;
  border: 1px solid #ccc;
  border-radius: rem(4);
  box-shadow: 0 0 rem(10) rgba(0, 0, 0, 0.1);

  input {
    display: none;
  }

  &.active {
    background: $active-color;
  }
}
