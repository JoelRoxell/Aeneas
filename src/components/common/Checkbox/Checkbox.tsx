import * as React from 'react'
import * as classNames from 'classnames'

import * as style from './Checkbox.scss'

interface CheckboxProps {
  onChange?: () => any
  selected: boolean
  name?: string
}

function Checkbox(props: CheckboxProps) {
  return (
    <div
      onClick={props.onChange}
      className={classNames(style.checkbox, {
        [style.active]: props.selected
      })}
    >
      <input
        className={style.checkbox}
        type="checkbox"
        checked={props.selected}
        onChange={props.onChange}
        name={props.name}
      />
    </div>
  )
}

export default Checkbox
