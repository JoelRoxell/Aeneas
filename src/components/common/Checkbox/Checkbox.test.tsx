import * as React from 'react'
import { mount } from 'enzyme'
import Checkbox from 'components/common/Checkbox'
import { act } from 'react-dom/test-utils'

const CheckboxWrapper = () => {
  const [active, setActive] = React.useState(false)

  return (
    <Checkbox
      selected={active}
      onChange={() => {
        console.log('Change!')

        setActive(!active)
      }}
    />
  )
}

describe('Checkbox', () => {
  test('render', () => {
    const wrapper = mount(<Checkbox selected />)

    expect(wrapper).toMatchSnapshot()
  })
  test('toggle', () => {
    const wrapper = mount(<CheckboxWrapper />)

    expect(wrapper.find(Checkbox).prop('selected')).toBeFalsy()

    act(() => {
      wrapper.find(Checkbox).simulate('click')
    })
    wrapper.update()

    expect(wrapper.find(Checkbox).prop('selected')).toBeTruthy()
  })
})
