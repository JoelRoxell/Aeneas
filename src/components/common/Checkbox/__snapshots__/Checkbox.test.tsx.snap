// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox render 1`] = `
<Checkbox
  selected={true}
>
  <div
    className="checkbox active"
  >
    <input
      checked={true}
      className="checkbox"
      type="checkbox"
    />
  </div>
</Checkbox>
`;
